### PR TITLE
External contract support

### PR DESCRIPTION
**Pull request by Uditgulati**

### CODE REVIEW

File: packages/junokit/src/lib/deploy/external_contract.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  ExternalContractsConfig,
+  JunokitRuntimeEnvironment,
+  StdFee,
+  UserAccount
+} from "../../types";
+import { loadCheckpoint, persistCheckpoint } from "../checkpoints";
+import { ExecuteResult, getClient, getSigningClient } from "../client";
+import { defaultFees } from "../contants";
+import { Abi, AbiParam } from "./abi";
+
```
Comment (amityadav0) [inline packages/junokit/src/lib/deploy/external_contract.ts:22]: Please remove warnings above.

**Review by amityadav0 — COMMENTED** [review]
Could you please add a doc about this feature ?

File: packages/junokit/sample-project/scripts/sample-script.js [hunk 2]
```diff
@@ -7,6 +7,32 @@ async function run() {
...
+  await junoswap_contract.setUpclient();
+
+  // query external contract
+  const ext_query_response = await junoswap_contract.query(
+    "balance",
+    {
+      address: contract_owner.account.address
+    }
+  );
+  console.log(ext_query_response);
```
Comment (amityadav0) [inline packages/junokit/sample-project/scripts/sample-script.js:20]: i think we should add this in junokit-templates and keep the initial boilerplate easier to understand.

Comment (Uditgulati) [reply packages/junokit/sample-project/scripts/sample-script.js:20]: sure, we can move it to a template which is more complex example. For example https://github.com/arufa-research/junokit-templates/issues/5

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**